File: pyNBA/scripts/_calib.py
# scripts/_calib.py
import json
import math
from calibration import build_calibration_table

def main():
    with open("data/history.json", "r") as f:
        h = json.load(f)

    # Check sResult values
    result_vals = set()
    buckets = {}

    for r in h.get("runs", []):
        if r.get("burnIn"):
            continue
        for g in r.get("games", []):
            s_pick = g.get("sPick")
            if not s_pick or s_pick == "PASS":
                continue
            s_result = g.get("sResult")
            if s_result:
                result_vals.add(s_result)
            if s_result not in ("WIN", "LOSS"):
                continue
            p = g.get("pCover", 0) or 0
            b = math.floor(p * 20) / 20
            key = f"{b:.2f}"
            if key not in buckets:
                buckets[key] = {"w": 0, "l": 0}
            if s_result == "WIN":
                buckets[key]["w"] += 1
            else:
                buckets[key]["l"] += 1

    print("sResult values:", sorted(result_vals))
    print("\n=== NBA P(cover) Calibration ===")
    print("Bucket    | W    | L    | Actual | Expected | D")
    for k in sorted(buckets.keys(), key=lambda x: float(x)):
        v = buckets[k]
        n = v["w"] + v["l"]
        actual = f"{v['w'] / n * 100:.1f}"
        expected = f"{float(k) * 100 + 2.5:.0f}"
        delta = f"{v['w'] / n * 100 - float(k) * 100 - 2.5:.1f}"
        print(f"P={k}  | {str(v['w']):<5}| {str(v['l']):<5}| {actual:<7}| {expected:<9}| {delta}")

    # Also try calibration module
    print("\n=== NBA Calibration Module Output ===")
    table = build_calibration_table(h)
    for row in table:
        if row["spread"]["n"] == 0:
            continue
        mid = round(row["midpoint"] * 100)
        win_pct = row["spread"].get("winPct")
        delta = win_pct - mid if win_pct is not None else None
        print(f"{row['label']:<10} n={str(row['spread']['n']):<5} actual={win_pct}%  expected={mid}%  D={delta}  units={row['spread']['units']}")

if __name__ == "__main__":
    main()

File: jsNBA/scripts/_calib.mjs
import fs from 'fs';
import { buildCalibrationTable } from './calibration.mjs';

const h = JSON.parse(fs.readFileSync('data/history.json','utf8'));

// Check sResult values
const resultVals = new Set();
let buckets = {};

for (const r of (h.runs || [])) {
  if (r.burnIn) continue;
  for (const g of (r.games || [])) {
    if (!g.sPick || g.sPick === 'PASS') continue;
    if (g.sResult) resultVals.add(g.sResult);
    if (g.sResult !== 'WIN' && g.sResult !== 'LOSS') continue;
    const p = g.pCover || 0;
    const b = Math.floor(p * 20) / 20;
    const key = b.toFixed(2);
    if (!buckets[key]) buckets[key] = { w: 0, l: 0 };
    if (g.sResult === 'WIN') buckets[key].w++;
    else buckets[key].l++;
  }
}

console.log('sResult values:', [...resultVals]);
console.log('\n=== NBA P(cover) Calibration ===');
console.log('Bucket    | W    | L    | Actual | Expected | Δ');
for (const [k, v] of Object.entries(buckets).sort((a,b) => a[0]-b[0])) {
  const n = v.w + v.l;
  const actual = (v.w / n * 100).toFixed(1);
  const expected = (parseFloat(k) * 100 + 2.5).toFixed(0);
  const delta = (v.w / n * 100 - parseFloat(k) * 100 - 2.5).toFixed(1);
  console.log(`P=${k}  | ${String(v.w).padEnd(5)}| ${String(v.l).padEnd(5)}| ${actual.padEnd(7)}| ${expected.padEnd(9)}| ${delta}`);
}

// Also try calibration module
console.log('\n=== NBA Calibration Module Output ===');
const table = buildCalibrationTable(h);
for (const r of table) {
  if (r.spread.n === 0) continue;
  const mid = Math.round(r.midpoint * 100);
  const delta = r.spread.winPct != null ? r.spread.winPct - mid : null;
  console.log(`${r.label.padEnd(10)} n=${String(r.spread.n).padEnd(5)} actual=${r.spread.winPct}%  expected=${mid}%  Δ=${delta}  units=${r.spread.units}`);
}
